### DISCUSSION TITLE
Select by marking the search options

### DISCUSSION THREAD

**Discussion by Curtixx** [wireui/wireui#956] [Q&A] [answered]
**To Reproduce**
1. put a wire:model in a select
2. put a :async-data in a select

**Dependencies**
-WireUI Version: 2.05
-Laravel Version: 10.4
-Livewire Version: 3.5
-Lang: pt_br

**Comment by ph7jack** [comment]
Hello @Curtixx, it seems to be working on our tests and on my local manual test. 
Could you please provide more information? A video, a screenshot, a code snippet...

**Comment by Curtixx** [comment]
- Essas são as categorias que o meu produto possui (isso esta no meu wire:model)
![image](https://github.com/user-attachments/assets/e6ef80fd-634a-4261-ab51-842d1b3f16f4)

- Porem no meu select ele esta marcando todas as opções de categoria que existe no sistema
![image](https://github.com/user-attachments/assets/230a21e5-9fed-4a44-a46d-b3aad9dad9e8)

- Eu tenho um :async-data que faz essa busca
![image](https://github.com/user-attachments/assets/ba30a645-510e-4c36-9324-474621e8febd)

- Assim que esta o meu select
![image](https://github.com/user-attachments/assets/458ae546-24a9-41fc-baa0-cb086256afa5)

**Comment by ph7jack** [comment, marked as answer]
I got it, it definitely is not an issue.

You aren't using the component properly.
1. Your model should be only the value you want, in this case an array of ids. `int[]`
2. Your search API is not properly set up, check the docs for more information on how to do it.
2.1. Please, read https://wireui.dev/components/select#async-search 
2.2. Remember that you need the [search](https://github.com/wireui/docs/blob/main/src/Examples/UserController.php#L17-L22) and the [selected](https://github.com/wireui/docs/blob/main/src/Examples/UserController.php#L23-L27) scopes to have it working. The select will search and fetch the results in the same API, if you return all categories, it will receive all categories as selected because the "selected" filter don't exists

Thanks for choosing wireui as your component's package. I'll convert this to a discussion so we can better guide it there.

**Comment by Curtixx** [comment]
Funcionou, muito obrigado pela atenção e desculpa qualquer coisa

**Reply by ph7jack** [reply]
That is no problem. I'm glad it has worked.
